dataprocxy/dataprocxy.py

```diff
@@ -71,53 +72,67 @@ def run(self):
         self.shutdown()
 
     def get_master_status(self, master_node, zone):
-        request = self.gce_service.instances().get(
-            project=self.project_id, zone=zone, instance=master_node)
-        response = request.execute()
-        return response['status'].encode('utf8')
+        try:
+            request = self.gce_service.instances().get(project=self.project_id, zone=zone, instance=master_node)
+            response = request.execute()
+            return response['status'].encode('utf8')
+        except HttpError as error:
+            print ('There was a Compute Engine API error when trying to query a Master node:\n %s, %s' %
+                   (error.resp.status, error._get_reason()))
+            exit(1)
 
     def query_cluster(self):
-        request = self.dataproc_service.projects().regions().clusters().get(
-            projectId=self.project_id,
-            region=self.region,
-            clusterName=self.cluster_name)
-        response = request.execute()
-        master_node = response['config']['masterConfig']['instanceNames'][0].encode('utf8')
-        zone = response['config']['gceClusterConfig']['zoneUri'].encode('utf8').split('/')[-1]
-        return master_node, zone
+        try:
+            request = self.dataproc_service.projects().regions().clusters().get(projectId=self.project_id,
+                                                                                region=self.region,
+                                                                                clusterName=self.cluster_name)
+            response = request.execute()
+            master_node = response['config']['masterConfig']['instanceNames'][0].encode('utf8')
+            zone = response['config']['gceClusterConfig']['zoneUri'].encode('utf8').split('/')[-1]
+            return master_node, zone
+        except HttpError as error:
+            self.handle_dataproc_http_error(error)
+
+    def get_cluster_from_job(self, job_id):
+        try:
+            request = self.dataproc_service.projects().regions().jobs().get(projectId=self.project_id,
+                                                                            region=self.region,
+                                                                            jobId=job_id)
+            response = request.execute()
+            cluster_name = response['placement']['clusterName'].encode('utf8')
+            return cluster_name
+        except HttpError as error:
+            self.handle_dataproc_http_error(error)
+
+    def handle_dataproc_http_error(self, error):
+        print ('There was a Cloud Dataproc API call error:\n %s, %s' % (error.resp.status, error._get_reason()))
+        if error.resp.status == 404:
+            print ('Perhaps you meant a different region than the one currently set:\n %s' % (self.region))
```

Comment:
I've been thinking about that, the problem is that it really just means that this particular cluster doesn't exist here, which can be a typo the wrong project or many other reasons why dataprocxy cannot find the cluster in this particular place. Maybe giving hints on what to check (is the cluster/job-id correct? did you specify the right project? have you set the correct region?) is more appropriate

Reply:
I agree, let me print out more info when encountering 404 response.